### PR TITLE
DOC: update the Period.qyear docstring

### PR DESCRIPTION
**Pull request by uttamaniket**
Checklist for the pandas documentation sprint (ignore this if you are doing
an unrelated PR):

 PR title is "DOC: update the <your-function-or-method> docstring"

 

- [x] The validation script passes: `scripts/validate_docstrings.py <your-function-or-method>`
- [x] The PEP8 style check passes: `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] The html version looks good: `python doc/make.py --single <your-function-or-method>`
- [ ] It has been proofread on language by another sprint participant

Please include the output of the validation script below between the "```" ticks:

```
(pandas_dev) ani@ani-Lenovo-G50-80:~/github/pandas/scripts$ python validate_docstrings.py  

- [ ] pandas.Period.qyear

################################################################################
####################### Docstring (pandas.Period.qyear)  #######################
################################################################################

Return a year of the given date.

This function simply find the year of the given date that period fallsin.

Returns
-------
int

See Also
--------
Period.qyear :Return the year of the date

Examples
--------
>>> p = pd.Period("2014-9-21", freq="A")
>>> p.qyear
2014

################################################################################
################################## Validation ##################################
################################################################################

```

If the validation script still gives errors, but you think there is a good reason
to deviate in this case (and there are certainly such cases), please state this
explicitly.

### CODE REVIEW

**Comment by uttamaniket** [comment]
@jerback please review the code

File: pandas/_libs/tslibs/period.pyx [hunk 1]
```diff
@@ -1408,6 +1408,25 @@ cdef class _Period(object):
 
     @property
     def qyear(self):
+        """
+        Return a year of the given date.
+
```
Comment (HimanshuAwasthi95) [inline pandas/_libs/tslibs/period.pyx]: year of the given date not appropriate . I'd say "Get the year component of the Period."

**Review by IHackPy — CHANGES_REQUESTED** [review]

File: pandas/_libs/tslibs/period.pyx [hunk 2]
```diff
@@ -1408,6 +1408,23 @@ cdef class _Period(object):
...
+        """
+        Get the year component of the Period.
+        
+        Returns
+        -------
+        int
+
+        See Also
+        --------
+        Period.qyear :Return the year of the date
```
Comment (HimanshuAwasthi95) [inline pandas/_libs/tslibs/period.pyx]: add pqyear in See Also section

Comment (TomAugspurger) [reply pandas/_libs/tslibs/period.pyx]: Can you fix the spacing here? Space after the `:`

**Review by TomAugspurger — COMMENTED** [review]
Can you remove the extra `.save` file from your PR?

**Comment by TomAugspurger** [comment]
Can you figure out how `pqyer` differs from `year`?

File: pandas/_libs/tslibs/period.pyx [hunk 3]
```diff
@@ -1408,6 +1408,24 @@ cdef class _Period(object):
...
+        """
+        Get the year component of the Period.
+
+        Returns
+        -------
+        int
+
+        See Also
+        --------
+        Period.qyear : Return the year of the date
```
Comment (jbrockmendel) [inline pandas/_libs/tslibs/period.pyx]: Isn't _this_ `Period.qyear`?  If so, it shouldn't be listed under See Also.

File: pandas/_libs/tslibs/period.pyx [hunk 4]
```diff
@@ -1408,6 +1408,24 @@ cdef class _Period(object):
 
     @property
     def qyear(self):
+        """
+        Get the year component of the Period.
+
+        Returns
+        -------
+        int
```
Comment (jbrockmendel) [inline pandas/_libs/tslibs/period.pyx]: `int` --> `year : int` (or `qyear : int`)

**Comment by uttamaniket** [comment]
@TomAugspurger  I am not getting difference between qyear and year. If any advice for me then please give me.

**Comment by TomAugspurger** [comment]
Not sure. Perhaps @jbrockmendel knows.

**Comment by jbrockmendel** [comment]
I think the distinction is for `freq`s where the calendar-quarter doesn't line up with the fiscal quarter.

```
per = pd.Period('2016Q1', freq='Q')
per2 = pd.Period('2016Q1', freq='Q-JUN')

>>> per.start_time
Timestamp('2016-01-01 00:00:00')

>>> per2.start_time
Timestamp('2015-07-01 00:00:00')

>>> per2.year
2015
>>> per2.qyear
2016
```

**Comment by uttamaniket** [comment]
Thanks @jbrockmendel for your advice.

**Comment by uttamaniket** [comment]
Please check the PR @jbrockmendel and give review.

File: pandas/_libs/tslibs/period.pyx [hunk 5]
```diff
@@ -1408,6 +1408,30 @@ cdef class _Period(object):
 
     @property
     def qyear(self):
+        """
+        Get the year component of the Period.
+
+        Returns
+        -------
+        int --> qyear : int
```
Comment (TomAugspurger) [inline pandas/_libs/tslibs/period.pyx]: This should just be `int` or `qyear : int`.

File: pandas/_libs/tslibs/period.pyx [hunk 6]
```diff
@@ -1408,6 +1408,30 @@ cdef class _Period(object):
 
     @property
     def qyear(self):
+        """
+        Get the year component of the Period.
```
Comment (TomAugspurger) [inline pandas/_libs/tslibs/period.pyx]: This should differentiate `.qyear` from just `.year`

File: pandas/_libs/tslibs/period.pyx [hunk 7]
```diff
@@ -1408,6 +1408,30 @@ cdef class _Period(object):
...
+        --------
+        Period.year : Return the year of the date
+        Period.pqyear : Return the year of the period
+
+        Examples
+        --------
+        >>> p = pd.Period('2014Q1', freq="Q")
+        >>> p.qyear
+        2014
+
```
Comment (TomAugspurger) [inline pandas/_libs/tslibs/period.pyx]: It'd be good to add a bit of text explaining the difference here. Something like

    The fiscal-quarter year can differ from the calendar year.

**Comment by jbrockmendel** [comment]
+1 to each of Tom's points.

**Comment by uttamaniket** [comment]
@TomAugspurger review it.

**Comment by datapythonista** [comment]
@jbrockmendel can you please review?

File: pandas/_libs/tslibs/period.pyx [hunk 8]
```diff
@@ -1463,6 +1463,45 @@ cdef class _Period(object):
 
     @property
     def qyear(self):
+        """
+        Get the fiscal year component of the Period.
+
+        The `year` and the `qyear` of the period will be the same if the fiscal
+        and the natural years are the same. When they are not, the fiscal year
```
Comment (jbrockmendel) [inline pandas/_libs/tslibs/period.pyx]: "natural"  --> "calendar"

File: pandas/_libs/tslibs/period.pyx [hunk 9]
```diff
@@ -1463,6 +1463,45 @@ cdef class _Period(object):
...
+        can be different from the natural year of the period.
+
+        Returns
+        -------
+        int
+
+        See Also
+        --------
+        Period.year : Return the natural year of the period.
```
Comment (jbrockmendel) [inline pandas/_libs/tslibs/period.pyx]: natural --> calendar

File: pandas/_libs/tslibs/period.pyx [hunk 10]
```diff
@@ -1463,6 +1463,45 @@ cdef class _Period(object):
...
+        If the natural and fiscal year are the same, `qyear` and `year` will
+        be the same.
+
+        >>> per = pd.Period('2018Q1', freq='Q')
+        >>> per.qyear
+        2018
+        >>> per.year
+        2018
+
+        If the fiscal year starts in April (`Q-MAR`), the first qearter of
```
Comment (jbrockmendel) [inline pandas/_libs/tslibs/period.pyx]: qearter --> quarter

File: pandas/_libs/tslibs/period.pyx [hunk 11]
```diff
@@ -1463,6 +1463,45 @@ cdef class _Period(object):
 
     @property
     def qyear(self):
+        """
+        Get the fiscal year component of the Period.
```
Comment (jbrockmendel) [inline pandas/_libs/tslibs/period.pyx]: Find the fiscal year the Period lies in according to the specified starting-quarter.

**Comment by datapythonista** [comment]
Thanks for the contribution @uttamaniket, and sorry for the delay in taking care of it.

**Pull request merged**